base/libgit2/tag.jl

```diff
@@ -35,6 +35,7 @@ function name(tag::GitTag)
 end
 
 function target(tag::GitTag)
-    oid_ptr = Ref(ccall((:git_tag_target_id, :libgit2), Ptr{Oid}, (Ptr{Void}, ), tag.ptr))
-    return oid_ptr[]
+    oid_ptr = ccall((:git_tag_target_id, :libgit2), Ptr{Oid}, (Ptr{Void}, ), tag.ptr)
+    oid_ptr == C_NULL && return Oid()
```

Comment:
the problem which would lead to this error would happen before this function call, so there is no point in investigation of this call problematic result, thus simple zero default will be returned. So, you start looking why zero id is returned, by looking at the tag creation functionality.